project_task_stage_change_restriction/models/project_task.py

```
@@ -0,0 +1,78 @@
+        for task in self:
+            if not self._is_move_allowed(task, new_stage, self.env.user):
+                raise UserError(
+                    _(
+                        "Sorry, you are not allowed to move the task "
+                        "'%(task)s' into the stage '%(stage)s'."
+                    )
+                    % {"task": task.display_name, "stage": new_stage.display_name}
+                )
+        return True
+
+    def write(self, vals):
```

Ruff (0.11.9)

Missing return type annotation for public function `write`

(ANN201)